### ISSUE TITLE
Server gives errors

### ISSUE THREAD

**Issue by dreynolds** [iancmcc/ouimeaux#116]
I tried this with the latest release and got similar errors, so I tried the HEAD of the repository and got the same thing:

`
Traceback (most recent call last):
 File "/home/pi/env/local/lib/python2.7/site-packages/gevent-1.1rc3-py2.7-linux-armv6l.egg/gevent/greenlet.py", line 522, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/pi/env/local/lib/python2.7/site-packages/gevent-1.1rc3-py2.7-linux-armv6l.egg/gevent/baseserver.py", line 153, in _close_when_done
    return handle(*args)
  File "/home/pi/env/local/lib/python2.7/site-packages/socketio/server.py", line 124, in handle
    handler.handle()
  File "/home/pi/env/local/lib/python2.7/site-packages/gevent-1.1rc3-py2.7-linux-armv6l.egg/gevent/pywsgi.py", line 418, in handle
    result = self.handle_one_request()
  File "/home/pi/env/local/lib/python2.7/site-packages/gevent-1.1rc3-py2.7-linux-armv6l.egg/gevent/pywsgi.py", line 633, in handle_one_request
    self.handle_one_response()
  File "/home/pi/env/local/lib/python2.7/site-packages/socketio/handler.py", line 122, in handle_one_response
    return self._do_handshake(handshake_tokens.groupdict())
  File "/home/pi/env/local/lib/python2.7/site-packages/socketio/handler.py", line 64, in _do_handshake
    self.write_smart(data)
  File "/home/pi/env/local/lib/python2.7/site-packages/socketio/handler.py", line 88, in write_smart
    self.write_plain_result(data)
  File "/home/pi/env/local/lib/python2.7/site-packages/socketio/handler.py", line 78, in write_plain_result
    ("Content-Type", "text/plain"),
  File "/home/pi/env/local/lib/python2.7/site-packages/gevent-1.1rc3-py2.7-linux-armv6l.egg/gevent/pywsgi.py", line 748, in start_response
    raise UnicodeError("The value must be a native string", header, value)
UnicodeError: ('The value must be a native string', 'Access-Control-Max-Age', 3600)
<Greenlet at 0x23cd580: _close_when_done(<socket at 0x244ebf0 fileno=[Errno 9] Bad file des,('192.168.2.55', 53016))> failed with UnicodeError`

Running on a Raspberry Pi using raspbian 7.8 with these devices
`(env)pi@raspberrypi ~/ouimeaux $ memo list
Bridge: WeMo Link
Light: Top landing
Light: Living room lamp
Switch: Living room`

**Comment by xfwhwh** [comment]
You may refer the answer from [#103](https://github.com/iancmcc/ouimeaux/issues/103) to solve this problem. 

> Steps on raspbian:
vi /usr/local/lib/python2.7/dist-packages/ouimeaux-0.7.9_r0-py2.7.egg/ouimeaux/subscribe.py
change line 76 from:
('Content-Length', len(SUCCESS)),
to:
('Content-Length', str(len(SUCCESS))),

**Comment by joshzitting** [comment]
this is still giving me issues on ubuntu 14.04 and ubuntu mate on a pi 3. 
thanks Love the API. hope we can get the server fixed!!

**Comment by joshzitting** [comment]
still get this error on my pi...
```python
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.1b4-py2.7-linux-armv7l.egg/gevent/greenlet.py", line 522, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/socketio/virtsocket.py", line 403, in _receiver_loop
    retval = pkt_ns.process_packet(pkt)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 155, in process_packet
    return self.process_event(packet)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 225, in process_event
    return self.call_method_with_acl(method_name, packet, *args)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 240, in call_method_with_acl
    return self.call_method(method_name, packet, *args)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 282, in call_method
    return method(*args)
  File "/usr/local/lib/python2.7/dist-packages/ouimeaux-0.7.9-py2.7.egg/ouimeaux/server/__init__.py", line 135, in on_join
    dispatch_uid=id(self))
TypeError: connect() got an unexpected keyword argument 'unique'
<Greenlet at 0x7573dd50: <bound method Socket._receiver_loop of <socketio.virtsocket.Socket object at 0x755d7eb0>>> failed with TypeError
```

**Comment by muldy** [comment]
Same here:
`Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.1.2-py2.7-linux-armv7l.egg/gevent/greenlet.py", line 534, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/socketio/virtsocket.py", line 403, in _receiver_loop
    retval = pkt_ns.process_packet(pkt)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 155, in process_packet
    return self.process_event(packet)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 225, in process_event
    return self.call_method_with_acl(method_name, packet, *args)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 240, in call_method_with_acl
    return self.call_method(method_name, packet, *args)
  File "/usr/local/lib/python2.7/dist-packages/socketio/namespace.py", line 282, in call_method
    return method(*args)
  File "/usr/local/lib/python2.7/dist-packages/ouimeaux/server/__init__.py", line 135, in on_join
    dispatch_uid=id(self))
TypeError: connect() got an unexpected keyword argument 'unique'
<Greenlet at 0x7548c2b0: <bound method Socket._receiver_loop of <socketio.virtsocket.Socket object at 0x75f0bbf0>>> failed with TypeError

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/gevent-1.1.2-py2.7-linux-armv7l.egg/gevent/greenlet.py", line 534, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2000, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1991, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python2.7/dist-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1567, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1642, in full_dispatch_request
    response = self.make_response(rv)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1731, in make_response
    raise ValueError('View function did not return a response')
ValueError: View function did not return a response
<Greenlet at 0x7548c260: <Flask 'ouimeaux.server'>({'wsgi.websocket': <geventwebsocket.websocket.WebS, <function <lambda> at 0x754d9cf0>)> failed with ValueError
`

**Comment by SethHaines** [comment]
Similar issue on CentOS 6.8 build  (Linux wemo-control 2.6.32-642.3.1.el6.x86_64 #1 SMP Tue Jul 12 18:30:56 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux) 

Running 0.8 of baseline with fix noted from #103 above. 

`Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/gevent/greenlet.py", line 534, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/lib/python2.6/site-packages/socketio/virtsocket.py", line 403, in _receiver_loop
    retval = pkt_ns.process_packet(pkt)
  File "/usr/lib/python2.6/site-packages/socketio/namespace.py", line 155, in process_packet
    return self.process_event(packet)
  File "/usr/lib/python2.6/site-packages/socketio/namespace.py", line 225, in process_event
    return self.call_method_with_acl(method_name, packet, *args)
  File "/usr/lib/python2.6/site-packages/socketio/namespace.py", line 240, in call_method_with_acl
    return self.call_method(method_name, packet, *args)
  File "/usr/lib/python2.6/site-packages/socketio/namespace.py", line 282, in call_method
    return method(*args)
  File "/usr/lib/python2.6/site-packages/ouimeaux/server/__init__.py", line 135, in on_join
    dispatch_uid=id(self))
TypeError: connect() got an unexpected keyword argument 'unique'
<Greenlet at 0x2d84af0: <bound method Socket._receiver_loop of <socketio.virtsocket.Socket object at 0x2ef7b90>>> failed with TypeError

Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/gevent/greenlet.py", line 534, in run
    result = self._run(*self.args, **self.kwargs)
  File "/usr/lib/python2.6/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib/python2.6/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python2.6/site-packages/flask_restful/__init__.py", line 271, in error_router
    return original_handler(e)
  File "/usr/lib/python2.6/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.6/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python2.6/site-packages/flask/app.py", line 1478, in full_dispatch_request
    response = self.make_response(rv)
  File "/usr/lib/python2.6/site-packages/flask/app.py", line 1566, in make_response
    raise ValueError('View function did not return a response')
ValueError: View function did not return a response
<Greenlet at 0x2d84a50: <Flask 'ouimeaux.server'>({'wsgi.websocket': <geventwebsocket.websocket.WebS, <function <lambda> at 0x2edfd70>)> failed with ValueError

`

strace information from when the web page is called 

`"", 128)                        = 0
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=15853, si_status=143, si_utime=0, si_stime=0} ---
wait4(-1, [{WIFEXITED(s) && WEXITSTATUS(s) == 143}], WNOHANG, NULL) = 15853
wait4(-1, 0x7fff7143d2dc, WNOHANG, NULL) = -1 ECHILD (No child processes)
rt_sigreturn()                          = 0
close(3)                                = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigaction(SIGINT, {0x43d610, [], SA_RESTORER, 0x391fa32660}, {SIG_DFL, [], SA_RESTORER, 0x391fa32660}, 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigaction(SIGINT, {SIG_DFL, [], SA_RESTORER, 0x391fa32660}, {0x43d610, [], SA_RESTORER, 0x391fa32660}, 8) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
read(255, "coproc mycoprocfd { \"${cmd[@]}\" "..., 384) = 322
pipe([3, 4])                            = 0
fcntl(63, F_GETFD)                      = -1 EBADF (Bad file descriptor)
dup2(3, 63)                             = 63
close(3)                                = 0
fcntl(63, F_GETFD)                      = 0
fcntl(62, F_GETFD)                      = -1 EBADF (Bad file descriptor)
dup2(4, 62)                             = 62
close(4)                                = 0
pipe([3, 4])                            = 0
fcntl(63, F_GETFD)                      = 0
fcntl(62, F_GETFD)                      = 0
fcntl(61, F_GETFD)                      = -1 EBADF (Bad file descriptor)
dup2(3, 61)                             = 61
close(3)                                = 0
fcntl(63, F_GETFD)                      = 0
fcntl(62, F_GETFD)                      = 0
fcntl(61, F_GETFD)                      = 0
fcntl(60, F_GETFD)                      = -1 EBADF (Bad file descriptor)
dup2(4, 60)                             = 60
close(4)                                = 0
rt_sigprocmask(SIG_BLOCK, [INT CHLD], [], 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [INT CHLD], 8) = 0
rt_sigprocmask(SIG_SETMASK, [INT CHLD], NULL, 8) = 0
lseek(255, -271, SEEK_CUR)              = 113
clone(child_stack=0, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f22f81b09d0) = 15872
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
close(62)                               = 0
close(61)                               = 0
fcntl(63, F_SETFD, FD_CLOEXEC)          = 0
fcntl(60, F_SETFD, FD_CLOEXEC)          = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [CHLD], 8) = 0
rt_sigprocmask(SIG_SETMASK, [CHLD], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [CHLD], 8) = 0
rt_sigprocmask(SIG_SETMASK, [CHLD], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [CHLD], 8) = 0
rt_sigprocmask(SIG_SETMASK, [CHLD], NULL, 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
read(255, "server_pid=$!\nread -t $timeout -"..., 384) = 271
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
fcntl(63, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
ioctl(63, SNDCTL_TMR_TIMEBASE or SNDRV_TIMER_IOCTL_NEXT_DEVICE or TCGETS, 0x7fff7143d8a0) = -1 EINVAL (Invalid argument)
lseek(63, 0, SEEK_CUR)                  = -1 ESPIPE (Illegal seek)
fstat(63, {st_mode=S_IFIFO|0600, st_size=0, ...}) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigaction(SIGALRM, {0x470580, [], SA_RESTORER, 0x391fa32660}, {SIG_DFL, [], 0}, 8) = 0
setitimer(ITIMER_REAL, {it_interval={0, 0}, it_value={4, 0}}, {it_interval={0, 0}, it_value={0, 0}}) = 0
read(63, "", 1)                         = 0
rt_sigaction(SIGALRM, {SIG_DFL, [], SA_RESTORER, 0x391fa32660}, {0x470580, [], SA_RESTORER, 0x391fa32660}, 8) = 0
setitimer(ITIMER_REAL, {it_interval={0, 0}, it_value={0, 0}}, {it_interval={0, 0}, it_value={3, 996141}}) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
fcntl(1, F_GETFD)                       = 0
fcntl(1, F_DUPFD, 10)                   = 10
fcntl(1, F_GETFD)                       = 0
fcntl(10, F_SETFD, FD_CLOEXEC)          = 0
dup2(2, 1)                              = 1
fcntl(2, F_GETFD)                       = 0
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 0), ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f22f81b9000
write(1, "Oh dear, the server with PID 158"..., 57Oh dear, the server with PID 15872 died after 4 seconds.
) = 57
dup2(10, 1)                             = 1
fcntl(10, F_GETFD)                      = 0x1 (flags FD_CLOEXEC)
close(10)                               = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
close(63)                               = 0
close(60)                               = 0
exit_group(1)                           = ?
+++ exited with 1 +++`
